src/include/txn.i

```diff
@@ -235,14 +235,11 @@ static inline void
 __wt_txn_unmodify(WT_SESSION_IMPL *session)
 {
 	WT_TXN *txn;
-	WT_TXN_OP *op;
 
 	txn = &session->txn;
 	if (F_ISSET(txn, WT_TXN_HAS_ID)) {
 		WT_ASSERT(session, txn->mod_count > 0);
-		op = txn->mod + txn->mod_count - 1;
-		__wt_txn_op_free(session, op);
-		txn->mod_count--;
+		__wt_txn_op_free(session, txn->mod + --txn->mod_count);
```

Comment:
I'm sure this is correct, but I find it hard to reason about. I don't really like getting array offsets via addition - which is why I preferred to make it explicit. I'd presumed the compiler was smart enough to get rid of the local variable for me.

Reply:
@agorrod, I'm sure you're correct, the compiler will kill the local store.

FTR, my reasoning is I don't like to see things done more than once, so the `x - 1` followed by `--x` triggered for me. I'm good either way if anyone else has a preference.